src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/SpecifiedResolverValueResolver.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Attribute\ValueResolver;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\ResolverNotFoundException;
+use Symfony\Contracts\Service\ServiceProviderInterface;
+
+/**
+ * Delegates value resolving of arguments bearing the {@see ValueResolver} attribute to the specified resolver.
+ *
+ * @author Mathieu Lechat <mathieu.lechat@les-tilleuls.com>
+ */
+final class SpecifiedResolverValueResolver implements ValueResolverInterface
+{
+    public function __construct(private readonly ContainerInterface $resolvers)
+    {
+    }
+
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+        if (!$attributes = $argument->getAttributesOfType(ValueResolver::class, ArgumentMetadata::IS_INSTANCEOF)) {
+            return [];
+        }
+
+        $resolverName = $attributes[0]->name;
+        if (!$this->resolvers->has($resolverName)) {
+            throw new ResolverNotFoundException($resolverName, $this->resolvers instanceof ServiceProviderInterface ? array_keys($this->resolvers->getProvidedServices()) : []);
+        }
+        $resolver = $this->resolvers->get($resolverName);
+        if (!$resolver instanceof ArgumentValueResolverInterface && !$resolver instanceof ValueResolverInterface) {
+            throw new \DomainException(sprintf('"%s" resolver must implement "%s" or "%s", but got "%s".', $resolverName, ArgumentValueResolverInterface::class, ValueResolverInterface::class, get_debug_type($resolver)));
+        }
+
+        $resolvedValue = $resolver->resolve($request, $argument);
+        if ($resolvedValue instanceof \Traversable) {
+            $resolvedValue = iterator_to_array($resolvedValue);
+        }
+        if (!$resolvedValue) {
+            throw new \UnexpectedValueException(sprintf('"%s" resolver did not yield any value for the "%s" argument.', $resolverName, $argument->getName()));
```

Comment:
if we remove this check, we can make MapEntity extend ValueResolver and this will work with `MapEntity(disabled: true)`

Reply:
I think one goal of `ValueResolver` was to ensure the value was resolved by the resolver you specified. This would not be the case anymore if we remove this check, would it?

Reply:
If `MapEntity` extends `ValueResolver`, doing `MapEntity(disabled: true)` would be very clear to the reader IMHO.

Reply:
That would work for the `MapEntity` case, but il would make it harder to debug that a faulty resolver doesn't properly resolve arguments when the attribute is here.